internal/test/e2e/validate_test.go

```diff
@@ -464,3 +519,45 @@ func (vm vmMetric) CompareTimestampValues(pmm *deployment.PMM, with vmMetric) in
 
 	return int(math.Abs(float64(len(xMap) - len(yMap))))
 }
+
+// Go usually can't handle floats with 16+ digits.
+// For example, when comparing them, the dump has the same numbers, but Go's float64 can interpret them as two different numbers:
+// 1) 0.00999923005927 and 0.00999923005928.
+// 2) 1.7540397181656854e+09 and 1.7540397181656857e+09.
+// 3) 9.223372036854776e+18 and 9.223372036854775e+18.
+// So, we convert them to Big.Float, round them, and then compare them.
+
+func roundFloatAndCheck(f1, f2 float64) bool {
+	// stop rounding and just compare if f1 has 1 digit only
+	s := strconv.FormatFloat(f1, 'e', precisionForRounding, bitsForRounding)
+	dotIndex := strings.Index(s, "e")
+	if dotIndex == -1 {
+		return f1 == f2
+	}
+
+	// round and compare
+	n1Rounded := new(big.Float).SetPrec(precisionForRounding).SetMode(big.ToNearestAway).SetFloat64(f1)
+	n2Rounded := new(big.Float).SetPrec(precisionForRounding).SetMode(big.ToNearestAway).SetFloat64(f2)
+
+	return n1Rounded.Cmp(n2Rounded) == 0
+}
+
+func saveTimestampToMap(timestamp int64, value float64, vm vmMetric, destination map[string]float64) {
```

Comment:
I don't like that the name of function doesn't correspond to it's functionality. Let's keep the entire workaround in a single separate function.